### PR TITLE
Configuration objects vs. Dictionary

### PR DESCRIPTION
**Pull request by blamarvt**
I agree with Mark that defaults are a good thing to have, so that a configuration file is not required. Making object allows for easy viewing of defaults:

$ pydoc stacktester.config

or, more specifically:

$ pydoc stacktester.config.NovaConfig
$ pydoc stacktester.config.GlanceConfig

Waldon, please don't taze me because this change-set has a bunch of different refactorings...I'm really bad about going off on a tangent. I'll split this up if you'd like :) Config changes in one and name changing  in another.

### CODE REVIEW

File: bin/stacktester [hunk 1]
```diff
@@ -1,24 +1,32 @@
...
 
 
-def parse_options_or_die():
+def parse_options():
     parser = optparse.OptionParser()
-    parser.add_option("-c", "--config", dest="config", metavar="FILE",
+    parser.add_option("-c", 
+                      "--config", 
+                      dest="config", 
+                      metavar="FILE",
```
Comment (bcwaldon) [inline bin/stacktester:22]: Unnecessary?

File: bin/stacktester [hunk 2]
```diff
@@ -1,24 +1,32 @@
...
                       help="Load configuration from FILE.",
                       default="etc/stacktester.cfg")
+    parser.add_option("-v",
+                      "--verbose",
+                      dest="verbose",
+                      action="store_true",
+                      help="Toggle verbose output.")
```
Comment (bcwaldon) [inline bin/stacktester:29]: Let's maybe not do this?

File: stacktester/tests/test_servers.py [hunk 3]
```diff
@@ -53,22 +53,22 @@ class ServersTest(unittest.TestCase):
...
-        self.os.nova_api.wait_for_server_status(serverid, 'ACTIVE')
+        self.os.nova.wait_for_server_status(serverid, 'ACTIVE')
 
         self.assertEqual('testserver', data['server']['name'])
 
-        response, body = self.os.nova_api.request(
+        response, body = self.os.nova.request(
             'DELETE',
-            '/servers/%s' % serverid, 
+            '/servers/%s' % serverid,
```
Comment (bcwaldon) [inline stacktester/tests/test_servers.py]: please put an underscore in serverid

File: stacktester/tests/test_servers.py [hunk 4]
```diff
@@ -53,22 +53,22 @@ class ServersTest(unittest.TestCase):
...
             body=body)
 
         # Raises TimeOutException on failure
-        self.os.nova_api.poll_request_status('GET', '/servers/%s' % serverid, 404)
+        self.os.nova.poll_request_status('GET', '/servers/%s' % serverid, 404)
```
Comment (bcwaldon) [inline stacktester/tests/test_servers.py]: please put an underscore in serverid

File: bin/stacktester (hunk 1 above)
Comment (blamarvt) [inline bin/stacktester:22]: Standard.

File: stacktester/tests/test_flavors.py [hunk 5]
```diff
@@ -47,13 +47,13 @@ class FlavorsTest(unittest.TestCase):
...
         url = '/flavors'
-        response, body = self.os.nova_api.request('GET', url)
+        response, body = self.os.nova.request('GET', url)
         self.assertEqual(response['status'], '200')
         actuals = json.loads(body)['flavors']
 
         for actual in actuals:
             print actual
-            
+
```
Comment (bcwaldon) [inline stacktester/tests/test_flavors.py:56]: please revert this change

File: bin/stacktester (hunk 2 above)
Comment (blamarvt) [inline bin/stacktester:29]: Uhh, I like seeing the test names as they run? Why not do this?

File: stacktester/nova.py [hunk 6]
```diff
@@ -18,39 +38,75 @@ class API(common.http.Client):
             print "Failed to authenticate user"
             raise
 
-        #TODO: use management_url
+        #TODO: use management_url <== what does this mean?
```
Comment (bcwaldon) [inline stacktester/nova.py]: use server_management_url returned from original authentication request

File: stacktester/tests/test_servers.py (hunk 3 above)
Comment (blamarvt) [inline stacktester/tests/test_servers.py]: Fixed.

File: bin/stacktester (hunk 2 above)
Comment (bcwaldon) [inline bin/stacktester:29]: I was referring to the line-breaking style change

File: stacktester/tests/test_servers.py (hunk 4 above)
Comment (blamarvt) [inline stacktester/tests/test_servers.py]: Fixed.

File: stacktester/tests/test_flavors.py (hunk 5 above)
Comment (blamarvt) [inline stacktester/tests/test_flavors.py:56]: <3
